packages/ui-tests/cypress/e2e/designer/catalogs/quarkusCatalog.cy.ts

```diff
@@ -0,0 +1,45 @@
+describe('Tests for Quarkus catalog type', () => {
+  beforeEach(() => {
+    cy.openHomePage();
+  });
+
+  it('Camel Quarkus catalog type with CR', () => {
+    cy.selectRuntimeVersion('Quarkus', 'Camel Quarkus 3.12.0');
+    cy.uploadFixture('flows/camelRoute/basic.yaml');
+    cy.openDesignPage();
+
+    cy.selectAppendNode('setHeader');
+    cy.checkCatalogVersion('3.12.0');
+    cy.chooseFromCatalog('component', 'as2');
+    cy.checkNodeExist('as2', 1);
+
+    cy.selectPrependNode('setHeader');
+    cy.checkCatalogVersion('3.12.0');
+    cy.chooseFromCatalog('processor', 'log');
+    cy.checkNodeExist('log', 2);
+
+    cy.openSourceCode();
+    cy.checkCodeSpanLine('uri: as2', 1);
+    cy.checkCodeSpanLine('log', 1);
+  });
+
+  it('Camel Quarkus catalog type with Kamelet', () => {
+    cy.selectRuntimeVersion('Quarkus', 'Camel Quarkus 3.12.0');
+    cy.uploadFixture('flows/kamelet/basic.yaml');
+    cy.openDesignPage();
+
+    cy.selectPrependNode('setBody');
+    cy.checkCatalogVersion('3.12.0');
```

Comment:
@tplevko do you think we can replace `3.12.0` with some RegExp? I don't think we can guarantee to have those versions always :cry:

Reply:
good point, replaced with a check, where we extract the current catalog version from the catalog version `data-testid`